### PR TITLE
[overlay] move segment_count_on_ring and segment_distance to get_ring, and add unit tests

### PR DESCRIPTION
**Pull request by barendgehrels**
This is a preparation step for next PR, which will be larger.
Because I need `segment_distance` for an update in turn clusters too.

This smaller PR is just a move, just refactoring.

### CODE REVIEW

File: test/count_set.hpp [hunk 1]
```diff
@@ -54,7 +52,7 @@ struct count_set
     friend std::ostream &operator<<(std::ostream &os, const count_set& s)
     {
        os << "{";
-       BOOST_FOREACH(std::size_t const& value, s.m_values)
+       for(std::size_t const& value : s.m_values)
```
Comment (barendgehrels) [inline test/count_set.hpp:55]: This is non related but I just got it in too

**Comment by barendgehrels** [comment]
Are you OK with this PR?

File: test/count_set.hpp (hunk 1 above)
Comment (vissarion) [reply test/count_set.hpp:55]: great! But there is a space missing after `for`

File: test/count_set.hpp [hunk 2]
```diff
@@ -84,7 +82,7 @@ private :
         else if (a.size() > 1 && b.size() == 1)
         {
             // One of them is optional, add the second
-            BOOST_FOREACH(std::size_t const& value, a)
+            for(std::size_t const& value : a)
```
Comment (vissarion) [inline test/count_set.hpp:85]: here too

**Review by vissarion — APPROVED** [review]
thanks! OK for merging!

**Review by awulkiew — APPROVED** [review]

File: test/count_set.hpp (hunk 1 above)
Comment (barendgehrels) [reply test/count_set.hpp]: :heavy_check_mark: thanks, fixed

File: test/count_set.hpp (hunk 2 above)
Comment (barendgehrels) [reply test/count_set.hpp]: :heavy_check_mark:

**Pull request merged**